server/notification/notification.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto2";
+option go_package = "github.com/argoproj/argo-cd/v2/pkg/apiclient/notification";
+
+// Notification Service
+//
+// Notification Service API performs query actions against notification resources
+package notification;
+
+import "google/api/annotations.proto";
+
+message Triggers {
+	repeated string triggers = 1;
+}
+
+message TriggersListRequest {}
+
+message Services {
+	repeated string services = 1;
+}
+
+message ServicesListRequest {}
+
+message Templates {
+	repeated string templates = 1;
+}
```

Comment:
Wondering if if it's short sighted for this API to return `[]string` or if it would be better to return `[]struct{name:string}`

Reply:
just in case? if we need something later?
what it can be for example?

Reply:
we can always provide more info in
```
/api/v1/notifications/triggers/name`
```
endpoint
@jessesuen

Reply:
For example, I believe some notifications allow for inputs from user. In addition to name, the API might provide metadata to prompt the UI to show the additional options for input (e.g. description or input name).

Reply:
@jessesuen done, pls check